### ISSUE TITLE
Improve transformer performance by skipping unnecessary work

### ISSUE THREAD

**Issue by cpw** [cpw/modlauncher#17]
See closed PR #16 for more discussion. I am not a fan of using null in this fashion however. I will investigate different ways to do this.

**Comment by ichttt** [comment]
I just watched some parts of your stream and want to clarify some things:
- The numbers I got where from running forge to the main menu with the code block I mentioned in my forge PR.
- There were actually two optimizations in my modlauncher PR: skipping the whole transformer step if only plugins are transforming (from [Line 52](https://github.com/cpw/modlauncher/blob/f9ca3bbc01022af18681446667df6f304cc78da1/src/main/java/cpw/mods/modlauncher/ClassTransformer.java#L52) to [Line 71](https://github.com/cpw/modlauncher/blob/f9ca3bbc01022af18681446667df6f304cc78da1/src/main/java/cpw/mods/modlauncher/ClassTransformer.java#L71)). This is badly readable in the git diff though.
- I think another optimisation with quite some impact would be to ask plugins for their needed ClassWriter flags, so if there is any transformer preset, use that COMPUTE_FRAMES, otherwise use the highest level that resulted from the plugins that actually wrote (as e.g. the `object_holder_definalize` needs no writing flag). Could be a new method on `ILaunchPluginService `

Anyways, thanks for doing this the proper way and making actual benchmarks. Hopefully I can join next stream
